vector/src/main/java/im/vector/util/RoomUtils.java

```diff
@@ -671,6 +677,36 @@ public void onClick(DialogInterface dialog, int which) {
                 .show();
     }
 
+    /**
+     *  When user wants to update bing rules, display a dialog in order to notify him that he have
+     *  to restart app to complete the updating of notification rules.
+     *
+     * @param context
+     */
+    public static void showBingRulesUpdatingDialog(final Context context) {
+        new AlertDialog.Builder(context)
+                .setMessage(R.string.room_settings_update_notification_rules)
+                .setPositiveButton(R.string.restart_app_now, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        if (null != EventStreamService.getInstance()) {
+                            EventStreamService.getInstance().stopNow();
+                        }
+                        Intent intent = new Intent(context, SplashActivity.class);
+                        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```

Comment:
You should use intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK) instead, to restart the whole app.